numpy/distutils/command/config.py

```diff
@@ -101,8 +101,10 @@ def _wrap_method(self, mth, lang, args):
         return ret
 
     def _compile (self, body, headers, include_dirs, lang):
-        return self._wrap_method(old_config._compile, lang,
-                                 (body, headers, include_dirs, lang))
+        src, obj = self._wrap_method(old_config._compile, lang,
+                                     (body, headers, include_dirs, lang))
+        self.temp_files.append(obj + '.d')
```

Comment:
Hmm well it looks like the important change occurred on May 1st in #9038 , here: https://github.com/numpy/numpy/pull/9038/files#diff-5eda0a2245b999f05879ef1ce9a3ce32R65

(which confuses me a little because I feel like I was having long build times before then... unfortunately I can't go back to check since my build state is different now. My builds are fast now in any case.).

There, the compiler object was changed to create the `.d` files, but didn't clean them up. In this PR am making the build object clean up those files for the compiler object. I admit that doesn't seem quite right, but I didn't see any better places and these files are already fiddly patches into distutils so I don't see a great harm. Only the build object has a dedicated cleanup function, `_clean`.